src/pages/library/Showcase.js

```diff
@@ -16,104 +18,337 @@ import * as Style from 'src/libs/style'
       boxShadow: Style.standardShadow
     }
   }, [
-    div({ style: { flex: 1, minWidth: 0, padding: '15px 20px' } }, [
-      div({ style: { color: colors.accent(), fontSize: 16, lineHeight: '20px', height: 40, marginBottom: 7 } }, [name]),
-      div({ style: { lineHeight: '20px', height: 100, whiteSpace: 'pre-wrap', overflow: 'hidden' } }, [description])
-    ]),
     div({
       style: {
-        backgroundRepeat: 'no-repeat', backgroundPosition: 'center', backgroundSize: 'auto 100%', borderRadius: '0 5px 5px 0',
+        backgroundRepeat: 'no-repeat', backgroundPosition: 'center', backgroundSize: 'auto 100%', borderRadius: '5px 0 0 5px',
         width: 87,
         ...Utils.cond(
           [name.toLowerCase().includes('covid'), () => ({ backgroundImage: `url(${covidBg})` })],
           [variant === 'gatk', () => ({ backgroundColor: '#333', backgroundImage: `url(${gatkLogo})`, backgroundSize: undefined })],
           () => ({ backgroundImage: `url(${featuredBg})`, opacity: 0.75 })
         )
       }
-    })
+    }),
+    div({ style: { flex: 1, minWidth: 0, padding: '15px 20px', overflow: 'hidden' } }, [
+      div({ style: { display: 'flex' } }, [
+        div({ style: { flex: 1, color: colors.accent(), fontSize: 16, lineHeight: '20px', height: 40, marginBottom: 7 } }, [name]),
+        created && div([Utils.makeStandardDate(created)])
+      ]),
+      div({ style: { lineHeight: '20px', height: 100, whiteSpace: 'pre-wrap', overflow: 'hidden' } }, [description])
+      // h(MarkdownViewer, [description]) // TODO: should we render this as markdown?
+    ])
   ])
 }
 
+// collapsible section for sidebar categories
+const SidebarCollapser = ({ title, isOpened, onClick, children }) => {
+  return div({
+    role: 'group'
+  }, [
+    h(NavItem, {
+      onClick,
+      style: { ...styles.sidebarRow, ...styles.nav.navSection }
+    }, [
+      title,
+      div({ style: { flexGrow: 1 } }),
+      icon(isOpened ? 'angle-up' : 'angle-down', { size: 18, style: { flex: 'none' } })
+    ]),
+    div({
+      style: { flex: 'none' }
+    }, [h(RCollapse, { isOpened }, [children])])
+  ])
+}
+
+const groupByFeaturedTags = workspaces => _.flow([
+  _.map(tag => [tag, _.filter(w => _.includes(tag, w.tags.items), workspaces)]),
+  _.fromPairs
+])(uniqueSidebarTags)
+
+const Sidebar = props => {
+  const { onSectionFilter, onTagFilter, sections, selectedSections, selectedTags, workspacesByTag } = props
+
+  // setup open-ness state for each sidebar section
+  const initialOpenState = _.fromPairs(_.map(section => [section.name, true], sidebarSections))
+  const [openState, setOpenState] = useState(initialOpenState)
+
+  const unionSectionWorkspaces = section => {
+    return _.uniq(_.flatMap(tag => workspacesByTag[tag], section.tags))
+  }
+
+  return div({ style: { display: 'flex', flexDirection: 'column' } }, [
+    _.map(section => {
+      return section.keepCollapsed ?
+        h(Clickable, {
+          key: section.name,
```

Comment:
Since the `key` property is usually reserved for when React is misbehaving, I wonder if we should have a convention around adding a comment about why it is needed when we use it?

Reply:
React also complains very loudly on the console if you omit `key`.

Reply:
Yes, I remember now. I actually had a utility to apply children (effectively extracting them from a list) just to avoid this warning. Please do disregard.